include/fastdds/dds/topic/TopicDataType.hpp

```diff
@@ -154,7 +130,7 @@ class TopicDataType
      * @param [in] force_md5 Force MD5 checking.
      * @return True if correct.
      */
-    FASTDDS_EXPORTED_API virtual bool getKey(
+    FASTDDS_EXPORTED_API virtual bool compute_key(
```

Comment:
Would be nice to add an overload of `compute_key` that allows computing the key from a SerializedPayload_t.

This would allow us to change the code [here](https://github.com/eProsima/Fast-DDS/blob/master/src/cpp/fastdds/subscriber/history/DataReaderHistory.cpp#L148-L154), avoiding the need of the `get_key_object_`.

Reply:
Done in 375996114